src/components/Admin/DropDown.jsx

```diff
@@ -1,58 +1,76 @@
 import React, { useState } from "react";
 import Dropdown from "react-bootstrap/Dropdown";
 import { RiArrowDownSLine } from "react-icons/ri";
-const Toggle = ({ children, onClick }) => {
+
+const Toggle = ({ children, onClick, show }) => {
   return (
     <button
       onClick={onClick}
       className="relative z-50 bg-white text-black w-full border-2 rounded-full border-black justify-between flex items-center px-3 py-1"
     >
       {children}
-      <RiArrowDownSLine />
+      <RiArrowDownSLine className={`${show && "rotate-180"} duration-300`} />
     </button>
   );
 };
 const Menu = ({ children, className, setOptions, options }) => {
+  const handleInput = (e) => {
+    setValue(e.target.value);
+    setOptions(
+      options.map((option) => ({
+        ...option,
+        hidden: !option.name.toLowerCase().includes(e.target.value),
+      }))
+    );
+  };
   const [value, setValue] = useState("");
   return (
     <div className={className}>
       <input
         autoFocus
-        className="mx-3 my-2 w-fit ring-0 outline-none"
+        className="mx-3 my-2 w-11/12 ring-0 !outline-0 border-gray-300 border-[1px] text-sm px-2 py-1"
         placeholder="Type to filter..."
-        onChange={(e) => {
-          setValue(e.target.value);
-          setOptions(
-            options.filter(
-              (option) => !value || option.toLowerCase().includes(value)
-            )
-          );
-        }}
+        onChange={handleInput}
         value={value}
       />
       {children}
     </div>
   );
 };
-const DropDown = ({ options, setOption, option, setOptions }) => {
+const DropDown = ({ options, setOptions, option, setOption }) => {
+  const [show, setShow] = useState(false);
   return (
-    <Dropdown className="w-full m-0 bg-white rounded-3xl">
-      <Dropdown.Toggle as={Toggle}>{option}</Dropdown.Toggle>
+    <Dropdown
+      onToggle={() => setShow(!show)}
+      show={show}
+      autoClose={true}
+      className="w-full m-0 bg-white rounded-3xl"
+    >
+      <Dropdown.Toggle show={show} as={Toggle}>
+        {option}
+      </Dropdown.Toggle>
       <Dropdown.Menu
         as={Menu}
-        setOptions={setOptions}
         options={options}
-        className="w-full bg-white border-none !rounded-b-xl !rounded-t-none border-2 border-black p-0 !z-10 !-mt-4 pt-4"
+        setOptions={setOptions}
+        className="max-h-[50vh] overflow-scroll w-full bg-white border-none !rounded-b-xl !rounded-t-none border-2 border-black p-0 !z-10 !-mt-4 pt-4"
       >
-        {options.map((currOoption, index) => (
-          <Dropdown.Item
-            className=" hover:!bg-hackathon-green-200 bg-transparent overflow-hidden px-3 py-1 last:rounded-b-xl"
-            key={index}
-            onClick={() => setOption(currOoption)}
-          >
-            {currOoption}
-          </Dropdown.Item>
-        ))}
+        {options.filter((opt) => !opt.hidden).length > 0 ? (
+          options.map(
+            (current, index) =>
```

Comment:
oh lol, hehe ma bad, but this should be option not current. I thought it was referring to a useState